tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_MinmodTci.cpp

```diff
@@ -729,11 +735,84 @@ void test_minmod_tci_3d() noexcept {
   test_activates(input, {{3.8, 2.1, 2.1, 2.7, 2.2, 2.5}}, {{0.0, 0.0, 0.0}});
 }
 
+struct ScalarTag : db::SimpleTag {
+  using type = Scalar<DataVector>;
+  static std::string name() noexcept { return "Scalar"; }
+};
+
+template <size_t VolumeDim>
+struct VectorTag : db::SimpleTag {
+  using type = tnsr::I<DataVector, VolumeDim>;
+  static std::string name() noexcept { return "Vector"; }
+};
+
+void test_minmod_tci_several_tensors() noexcept {
+  INFO("Testing MinmodTci action on several tensors");
+  // Test that TCI returns true if just one component needs limiting, which
+  // we can adequately test in 1D.
+  const auto minmod_type = SlopeLimiters::MinmodType::LambdaPi1;
+  const double tvbm_constant = 0.0;
+  const auto element = make_element<1>();
+  const size_t number_of_grid_points = 2;
+  const Mesh<1> mesh(number_of_grid_points, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto);
+  const auto element_size = make_array<1>(1.0);
+
+  Scalar<DataVector> local_scalar{DataVector{{1.4, 2.8}}};
+  tnsr::I<DataVector, 1> local_vector{DataVector{{-0.3, -0.4}}};
+
+  struct TestPackagedData {
+    tuples::TaggedTuple<::Tags::Mean<ScalarTag>, ::Tags::Mean<VectorTag<1>>>
+        means;
+    std::array<double, 1> element_size =
+        make_array<1>(std::numeric_limits<double>::signaling_NaN());
+  };
+  std::unordered_map<std::pair<Direction<1>, ElementId<1>>, TestPackagedData,
+                     boost::hash<std::pair<Direction<1>, ElementId<1>>>>
+      neighbor_data{};
+  TestPackagedData& lower_neighbor =
+      neighbor_data[std::make_pair(Direction<1>::lower_xi(), ElementId<1>(1))];
+  TestPackagedData& upper_neighbor =
+      neighbor_data[std::make_pair(Direction<1>::upper_xi(), ElementId<1>(2))];
+  lower_neighbor.element_size = element_size;
+  upper_neighbor.element_size = element_size;
+
+  // Case where neither tensor triggers limiting
+  get(get<::Tags::Mean<ScalarTag>>(lower_neighbor.means)) = 0.9;
+  get(get<::Tags::Mean<ScalarTag>>(upper_neighbor.means)) = 3.0;
+  get<0>(get<::Tags::Mean<VectorTag<1>>>(lower_neighbor.means)) = 0.1;
+  get<0>(get<::Tags::Mean<VectorTag<1>>>(upper_neighbor.means)) = -0.5;
+  bool activation = SlopeLimiters::Minmod_detail::troubled_cell_indicator<
+      1, TestPackagedData, ScalarTag, VectorTag<1>>(
+      local_scalar, local_vector, neighbor_data, minmod_type, tvbm_constant,
+      element, mesh, element_size);
+  CHECK_FALSE(activation);
+
+  // Case where the scalar triggers limiting
+  get(get<::Tags::Mean<ScalarTag>>(upper_neighbor.means)) = 2.0;
+  activation = SlopeLimiters::Minmod_detail::troubled_cell_indicator<
+      1, TestPackagedData, ScalarTag, VectorTag<1>>(
+      local_scalar, local_vector, neighbor_data, minmod_type, tvbm_constant,
+      element, mesh, element_size);
+  CHECK(activation);
+
+  // Case where the vector triggers limiting
```

Comment:
It would be good to test that any of the 3 components of the vector cause the limiter to trigger